src/course-unit/add-component/AddComponent.jsx

```diff
@@ -1,15 +1,34 @@
+import PropTypes from 'prop-types';
 import { useSelector } from 'react-redux';
+import { useNavigate } from 'react-router-dom';
 import { useIntl } from '@edx/frontend-platform/i18n';
 import { Button } from '@edx/paragon';
 
 import { getCourseSectionVertical } from '../data/selectors';
+import { COMPONENT_ICON_TYPES } from '../constants';
 import ComponentIcon from './ComponentIcon';
 import messages from './messages';
 
-const AddComponent = () => {
+const AddComponent = ({ blockId, handleCreateNewCourseXblock }) => {
+  const navigate = useNavigate();
   const intl = useIntl();
   const { componentTemplates } = useSelector(getCourseSectionVertical);
 
+  const handleCreateNewXblock = (type) => () => {
+    switch (type) {
+    case COMPONENT_ICON_TYPES.discussion:
+    case COMPONENT_ICON_TYPES.dragAndDrop:
+      handleCreateNewCourseXblock({ type, parentLocator: blockId });
+      break;
+    case COMPONENT_ICON_TYPES.problem:
+      handleCreateNewCourseXblock({ type, parentLocator: blockId }, ({ courseKey, locator }) => {
+        navigate(`/course/${courseKey}/editor/problem/${locator}`);
```

Comment:
[important] cover by util for path transform (adding `course-authoring/` at the beginning)